### PR TITLE
Substitute config values after parsing YAML

### PR DESCRIPTION
**Pull request by henryptung**
Instead of filtering and replacing in the raw JSON stream, filter and replace after parsing using a JSON tree visitor.

Avoids escaping concerns due to string replacement in the serialized form (i.e. YAML).

### CODE REVIEW

File: encrypted-config-value-bundle/src/main/java/com/palantir/config/crypto/jackson/JsonNodeStringReplacer.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            String field = entry.getKey();
+            JsonNode node = entry.getValue();
+            newObjectNode.set(field, JsonNodeVisitors.dispatch(node, this));
+        }
+        return newObjectNode;
+    }
+
+    @Override
+    public JsonNode visitPojo(POJONode pojoNode) {
+        // Anything to do here?
```
Comment (bavardage) [inline encrypted-config-value-bundle/src/main/java/com/palantir/config/crypto/jackson/JsonNodeStringReplacer.java:90]: is there anything to do here?

Comment (bavardage) [inline encrypted-config-value-bundle/src/main/java/com/palantir/config/crypto/jackson/JsonNodeStringReplacer.java:90]: (what's pojo node vs object node?)

**Comment by bavardage** [comment]
this looks good - just want to add in some tests (which I'm doing now)

do you have an example of where it would fail before (so I can be sure to test that case)

**Comment by dtolnay** [comment]
One example that would fail before is:

``` yaml
x: ${enc:...}
```

where the encrypted value is `%` so it expands out to this invalid YAML before parsing:

``` yaml
x: %
```

**Comment by bavardage** [comment]
cool, have pushed up some tests and created #27

**Comment by bavardage** [comment]
will merge #27

**Pull request closed without merging**